themes/default/content/blog/disabledefaultproviders/index.md

```diff
@@ -0,0 +1,280 @@
+const bucket = new aws.s3.Bucket("my-bucket", {
+  website: {
+    indexDocument: "index.html",
+  },
+}, { provider: west });
+
+// Inherit provider `west` from parent `bucket`
+const bucketObject = new aws.s3.BucketObject("index.html", {
+  acl: "public-read",
+  contentType: "text/html",
+  bucket: bucket,
+  source: new pulumi.asset.FileAsset("index.html"),
+}, { parent: bucket });
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```python
+west = aws.Provider("provider", region="us-east-1")
+
+bucket = aws.s3.Bucket(
+    "my-bucket",
+    website=aws.s3.BucketWebsiteArgs(index_document="index.html"),
+    opts=ResourceOptions(provider=west),
+)
+
+bucketObject = aws.s3.BucketObject(
+    "index.html",
+    acl="public-read",
+    content_type="text/html",
+    bucket=bucket,
+    source=asset.FileAsset("index.html"),
+    opts=ResourceOptions(parent=bucket),
+)
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+west, err := aws.NewProvider(ctx, "provider", &aws.ProviderArgs{
+	Region: aws.RegionUSEast1,
+})
+if err != nil {
+	return err
+}
+
+bucket, err := s3.NewBucket(ctx, "my-bucket", &s3.BucketArgs{
+	Website: s3.BucketWebsiteArgs{
+	IndexDocument: pulumi.StringPtr("index.html"),
+	},
+}, pulumi.Provider(west))
+if err != nil {
+	return err
+}
+
+_, err = s3.NewBucketObject(ctx, "index.html", &s3.BucketObjectArgs{
+	Acl:         s3.CannedAclPublicRead,
+	ContentType: pulumi.StringPtr("text/html"),
+	Bucket:      bucket,
+	Source:      pulumi.NewFileAsset("index.html"),
+}, pulumi.Parent(bucket))
+if err != nil {
+	return err
+}
+```
+
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+
+```csharp
+var west = new Aws.Provider("provider", new Aws.ProviderArgs {
+        Region = "us-east-1",
+});
+
+var bucket = new S3.Bucket("my-bucket", new S3.BucketArgs {
+        Website = new Aws.S3.Inputs.BucketWebsiteArgs {
+            IndexDocument = "index.html"
+        },
+    },
+    new CustomResourceOptions { Provider = west },
+);
+
+new S3.BucketObject("index.html", new S3.BucketObjectArgs {
+        Acl = "public-read",
+        ContentType = "text/html",
+        Bucket = bucket,
+        Source = FileAsset("index.html"),
+     },
+    new CustomResourceOptions { Parent = bucket },
+);
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+Default providers help keep simple things simple but make other actions impossible. As demonstrated in the above example, AWS providers, whether they are explicit or default, control the deployment region. That means multi-region deployments necessitate explicit providers. Likewise, deploying to a newly created Kubernetes cluster requires explicit providers since Kubernetes providers describe their cluster. There are many scenarios when explicit providers are mandatory for a correct deployment. When it is important that only explicitly configured providers are used, default providers lead to unpredictable deployments.
```

Comment:
>When it is important that only explicitly configured providers are used, default providers lead to unpredictable deployments.

"Unpredictable" sounds a bit scary. Is it accurate? I can see how mixing explicit and default providers within the same program could lead to confusion or surprise, but not so much unpredictability. (Unless I'm misunderstanding of course.) Would it still be accurate to put it this way instead?

```suggestion
Default providers help keep simple things simple but make other actions impossible. As demonstrated in the above example, AWS providers, whether they are explicit or default, control the deployment region. That means multi-region deployments necessitate explicit providers. Likewise, deploying to a newly created Kubernetes cluster requires explicit providers since Kubernetes providers describe their cluster. There are many scenarios when explicit providers are mandatory for a correct deployment. When it is important that only explicitly configured providers are used, default providers can lead to unexpected results.
```